streams/src/main/java/org/apache/kafka/streams/processor/TopologyDescription.java

```diff
@@ -0,0 +1,412 @@
+        public final int id;
+        /** All nodes of this sub-topology. */
+        public final Set<Node> nodes;
+
+        Subtopology(final int id,
+                    final Set<Node> nodes) {
+            this.id = id;
+            this.nodes = nodes;
+        }
+
+        @Override
+        public String toString() {
+            return "Sub-topology: " + id + "\n" + nodes();
+        }
+
+        private String nodes() {
+            final StringBuilder sb = new StringBuilder();
+            for (final Node node : nodes) {
+                sb.append("    ");
+                sb.append(node);
+                sb.append('\n');
+            }
+            return sb.toString();
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final Subtopology that = (Subtopology) o;
+            return id == that.id
+                && nodes.equals(that.nodes);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = id;
+            result = 31 * result + nodes.hashCode();
+            return result;
+        }
+    }
+
+    public final static class GlobalStore {
+        /** The name of the global store. */
+        public final Source source;
+        /** The source topic of the global store. */
+        public final Processor processor;
+
+        GlobalStore(final String sourceName,
+                    final String processorName,
+                    final String storeName,
+                    final String topicName) {
+            source = new Source(sourceName, topicName);
+            processor = new Processor(processorName, Collections.singleton(storeName));
+            source.successors.add(processor);
+            processor.predecessors.add(source);
+        }
+
+        @Override
+        public String toString() {
+            return "GlobalStore: " + source.name + "(topic: " + source.topics + ") -> "
+                + processor.name + "(store: " + processor.stores.iterator().next() + ")\n";
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            final GlobalStore that = (GlobalStore) o;
+            return source.equals(that.source)
+                && processor.equals(that.processor);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = source.hashCode();
+            result = 31 * result + processor.hashCode();
+            return result;
+        }
+    }
+
+    /**
+     * A node of a topology. Can be a source, sink, or processor node.
+     */
+    public interface Node {
+        /** The name of the node. */
```

Comment:
According to `TopologyBuilder` empty string would be ok -- `null` is not allowed. But do we need to document this here, too? Or even enforce in the constructor (that is not public) ?